router/router.go

```diff
@@ -1956,6 +1954,7 @@ func (rt *HandleT) readAndProcess() int {
 
 	rt.throttledUserMap = make(map[string]struct{})
 	throttledAtTime := time.Now()
+	reportMetrics := make([]*utilTypes.PUReportedMetric, 0)
```

Comment:
it would be better to limit the scope of this variable inside the `if drain {}` branch

Reply:
reportMetrics aggregates all the metric for all jobs and it has a scope outside not only the if condition, but the surrounding **for** loop